### DISCUSSION TITLE
Shareable knowledge between multiple users

### DISCUSSION THREAD

**Discussion by EricLiclair** [khoj-ai/khoj#459] [Ideas]
Pretty visionary but I'd love to see a shareable knowledge base between users. Something like an extendable community where every member polls to involve another person in there knowledge base.

We'd want to test this between two users for a test run. And then extend to a community level.

**Comment by debanjum** [comment]
Sounds like an interesting idea. Could you describe the use-case a little more? How do you envision sharing access to 
your personal knowledge base? What example scenarios do you imagine this will help with?

Edit: Converting this into a discussion for now

**Reply by EricLiclair** [reply]
Yeah. 
Say, 
User-1 and User-2 have their own knowledge bases that is used to answer their questions.
What if there is a way to allow User-2 to also access User-1's knowledge base, and vice-versa (maybe a "friend"-ly connection)

For example, I created book summaries for books A, B, C, D. And my friend read books G, H, I, J and wrote the summaries. I want to add him as a friend and be able to query on a shared knowledge base. To fetch a summary of what all books he has read, the summaries, etc. 
(Sounds like querying someone's brain 😂)

Similarly I might also be able to find and download images that have me in them from my friends' knowledge base. (Again visionary but achievable; face recog. and creating a metadata for the image database, a query to return all images with me in them, list them, allow me to download whatever I want. etc etc.)

And extending this slowly for other features 🤔.

What do you think?

**Reply by TomLucidor** [reply]
Remote drives and shared storage as key to communal KM, either SyncTrayzor for smaller projects or cloud storage for bigger affairs.
However packing the conversational training with chatbots would be something different requiring shared access.

**Reply by debanjum** [reply]
@EricLiclair neat idea! While Khoj isn't built to specifically support this use-case yet. The initial ideas you'd want should be already kind of possible. The setup would look like what @TomLucidor mentioned, you and friend(s) setup a shared local drive (e.g Syncthing, Dropbox etc.) that each of your Khoj is made to index, whenever you want to share stuff with the friend(s) you add it to this shared drive. Now you and your friends can search and chat with these shared notes using Khoj. You can use [file filters](https://docs.khoj.dev/#/advanced?id=query-filters) to limit to querying only the shared drive or your own notes.

**Reply by TomLucidor** [reply]
Mix in fuzzy search as well if you want a little flexibility, or embedding if accuracy is prioritized.
